src/CSModule.sol

```diff
@@ -1777,23 +1296,98 @@ contract CSModule is
+            uint32 enqueuedSoFar = no.totalDepositedKeys + no.enqueuedCount;
+
+            if (maxDeposits > enqueuedSoFar) {
+                uint32 leftForQueue = maxDeposits - enqueuedSoFar;
+                _enqueueNodeOperatorKeys(
+                    nodeOperatorId,
+                    priority,
+                    leftForQueue
+                );
+                no.usedPriorityQueue = true;
+            }
         }
+
+        _enqueueNodeOperatorKeys(
+            nodeOperatorId,
+            QUEUE_LOWEST_PRIORITY,
+            type(uint32).max
+        );
     }
 
-    function _onlyNodeOperatorManager(uint256 nodeOperatorId) internal view {
+    function _enqueueNodeOperatorKeys(
+        uint256 nodeOperatorId,
+        uint256 queuePriority,
+        uint32 maxKeys
+    ) internal {
         NodeOperator storage no = _nodeOperators[nodeOperatorId];
-        if (no.managerAddress == address(0)) revert NodeOperatorDoesNotExist();
-        if (no.managerAddress != msg.sender) revert SenderIsNotEligible();
+        uint32 depositable = no.depositableValidatorsCount;
+        uint32 enqueued = no.enqueuedCount;
+
+        if (enqueued < depositable) {
+            uint32 count;
+
+            unchecked {
+                count = depositable - enqueued;
+                if (count > maxKeys) count = maxKeys;
+                no.enqueuedCount = enqueued + count;
+            }
+
+            QueueLib.Queue storage q = _getQueue(queuePriority);
+            q.enqueue(nodeOperatorId, count);
+            emit BatchEnqueued(queuePriority, nodeOperatorId, count);
+        }
     }
 
-    function _onlyNodeOperatorManagerOrRewardAddresses(
-        uint256 nodeOperatorId
+    /// @dev Acts as a proxy to `_queueByPriority` till `legacyQueue` deprecation.
+    /// @dev TODO: Remove in CSM v3.
+    function _getQueue(
+        uint256 priority
+    ) internal view returns (QueueLib.Queue storage q) {
+        if (priority == QUEUE_LEGACY_PRIORITY) {
+            assembly {
+                q.slot := legacyQueue.slot
+            }
+        } else {
+            q = _queueByPriority[priority];
+        }
+    }
+
+    function _checkCanAddKeys(
+        uint256 nodeOperatorId,
+        address who
+    ) internal view {
+        // Most likely a direct call, so check the sender is a manager.
+        if (who == msg.sender) {
+            _onlyNodeOperatorManager(nodeOperatorId, msg.sender);
+        } else {
+            // We're trying to add keys via gate, check if we can do it.
+            _checkRole(CREATE_NODE_OPERATOR_ROLE);
+            if (_nodeOperators[nodeOperatorId].totalAddedKeys > 0) {
+                revert NodeOperatorHasKeys();
+            }
+        }
+    }
+
+    function _onlyNodeOperatorManager(
+        uint256 nodeOperatorId,
+        address from
     ) internal view {
         NodeOperator storage no = _nodeOperators[nodeOperatorId];
         if (no.managerAddress == address(0)) revert NodeOperatorDoesNotExist();
```

Comment:
Is it case possible?

Reply:
Yes. For the "non-created" node operators, the storage will be empty. Hence, no.managerAddress will be address(0)